cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  END_TIME,
+  START_TIME,
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+  INDEX_WITH_TIME_2,
+} from './constants';
+import { SECONDARY_ENGINE } from '../../../../../utils/constants';
+import { v4 as uuid } from 'uuid';
+
+const workspaceName = uuid();
+// datasource name must be 32 char or less
+const datasourceName = uuid().substring(0, 32);
+
+const allowedSearchOperations = {
+  DQL: {
+    filters: true,
+    histogram: true,
+    selectFields: true,
+    sort: true,
+  },
+  Lucene: {
+    filters: true,
+    histogram: true,
+    selectFields: true,
+    sort: true,
+  },
+  'OpenSearch SQL': {
+    filters: false,
+    histogram: false,
+    selectFields: true,
+    sort: false,
+  },
+  PPL: {
+    filters: false,
+    histogram: true,
+    selectFields: true,
+    sort: false,
+  },
+};
+
+const indexPatternTestConfigurations = {
+  DQL: {
+    ...allowedSearchOperations.DQL,
+    language: 'DQL',
+    dataset: INDEX_PATTERN_WITH_TIME,
+    queryString: 'bytes_transferred > 9950',
+    hitCount: 28,
+    sampleTableData: [
```

Comment:
nit: since we capsuled a lot of test items in, some item like sampleTableData is not quite clear to me. shall we add a comment at the beginning of indexPatternTestConfigurations.

Reply:
yeah can do

Reply:
added some notes and moved this to another place